src/pubkey.h

```diff
@@ -142,6 +142,9 @@ class CPubKey
         unsigned int len = ::ReadCompactSize(s);
         if (len <= SIZE) {
             s.read((char*)vch, len);
+            if (len != size()) {
+                Invalidate();
```

Comment:
Instead of enumerating what could go wrong, what do you think about having one path for a valid deserialization and let all other paths end in an invalid deserialization?


The diff would be:

```diff
diff --git a/src/pubkey.h b/src/pubkey.h
index 261842b7f7..0f4787bc90 100644
--- a/src/pubkey.h
+++ b/src/pubkey.h
@@ -142,13 +142,14 @@ public:
         unsigned int len = ::ReadCompactSize(s);
         if (len <= SIZE) {
             s.read((char*)vch, len);
+            if (len == size()) return;
         } else {
             // invalid pubkey, skip available data
             char dummy;
             while (len--)
                 s.read(&dummy, 1);
-            Invalidate();
         }
+        Invalidate();
     }
 
     //! Get the KeyID of this public key (hash of its serialization)